website/project/model.py

```diff
@@ -2359,15 +2371,19 @@ def add_log(self, action, params, auth, foreign_user=None, log_date=None, save=T
             user=user,
             foreign_user=foreign_user,
             params=params,
+            node=self,
+            original_node=params['node']
```

Comment:
Is it possible that neither `'node'` nor `'project'` would be in `params`, and if so, should this default to `self`, as above?